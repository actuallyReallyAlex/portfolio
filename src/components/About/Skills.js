import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading } from 'grommet'
import { about } from '../../config.json'
import SkillBox from './SkillBox.js'

const Skills = ({ size }) => (
  <Box
    align={size === 'small' ? 'center' : 'start'}
    direction={size === 'small' ? 'column' : 'row'}
    fill="horizontal"
    gap={size === 'small' ? 'large' : 'xlarge'}
    justify="center"
    margin={{ vertical: 'large' }}
  >
    {about.skills.map(({ sectionTitle, languages }, i) => (
      <Box align="start" key={i} justify="start">
        <Box align="center" margin={{ bottom: 'small' }}>
          <Heading level="3" margin={{ bottom: 'xsmall' }}>
            {sectionTitle}
          </Heading>
          <Box
            background="#eb1736"
            fill="horizontal"
            height="3px"
            round="small"
          />
        </Box>
        {languages.map(({ color, title }, i) => (
          <SkillBox color={color} key={i} title={title} />
        ))}
      </Box>
    ))}
  </Box>
)

Skills.propTypes = {
  size: PropTypes.string.isRequired
}

export default Skills
