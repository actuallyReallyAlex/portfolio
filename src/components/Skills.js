import React from 'react'
import { Box, Heading, Label } from 'grommet'
import LoremIpsum from './LoremIpsum'
import LanguagesLegend from './LanguagesLegend'

const Skills = () => {
  return (
    <Box>
      <Box direction="row" justify="center">
        <Box pad={{ vertical: 'small' }}>
          <Heading tag="h3">Skills</Heading>
          <LoremIpsum />
        </Box>
      </Box>
      <Box direction="row">
        <Box pad="large">
          <Label>Programming Languages</Label>
          <LanguagesLegend languages={['JavaScript', 'C#']} />
        </Box>
        <Box pad="large">
          <Label>Not Exactly A Programming Language</Label>
          <LanguagesLegend
            languages={['HTML', 'CSS', 'Markdown', 'MDX', 'JSX', 'JSON']}
          />
        </Box>
        <Box pad="large">
          <Label>Technologies / Frameworks</Label>
          <LanguagesLegend
            languages={['React.js', 'Node.js', 'JQuery', 'Bootstrap']}
          />
        </Box>
        <Box pad="large">
          <Label>Tools / Software</Label>
          <LanguagesLegend
            languages={[
              'MS Visual Studio',
              'Adobe Photoshop',
              'Adobe Illustrator'
            ]}
          />
        </Box>
      </Box>
    </Box>
  )
}

export default Skills
