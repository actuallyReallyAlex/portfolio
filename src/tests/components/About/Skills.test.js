import React from 'react'
import Skills from '../../../components/About/Skills'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<Skills />', () => {
  test('Should render the <Skills />.', () => {
    const component = shallow(<Skills {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
