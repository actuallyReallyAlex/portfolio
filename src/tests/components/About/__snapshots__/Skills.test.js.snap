// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> Should render the <Skills />. 1`] = `ShallowWrapper {}`;
